src/inngest/functions/sms/bulkSMSCommunicationJourney.ts

```diff
@@ -0,0 +1,356 @@
+    if (!DATABASE_QUERY_LIMIT || phoneNumbers.length < DATABASE_QUERY_LIMIT) {
+      break
+    }
+  }
+
+  // We need to simplify this array so that the payload doesn't exceed the 4MB limit that inngest has
+  return [
+    chunk(
+      phoneNumberList.map(({ phoneNumber }) => phoneNumber),
+      QUEUING_THROUGHPUT,
+    ),
+    innerCursor,
+  ]
+}
+
+const ENQUEUE_MAX_RETRY_ATTEMPTS = 5
+
+async function enqueueMessages(phoneNumbers: string[], body: string, attempt = 0) {
+  if (attempt > ENQUEUE_MAX_RETRY_ATTEMPTS) return 0
+
+  const enqueueMessagesPromise = phoneNumbers.map(async phoneNumber => {
+    const communicationJourneys = await createCommunicationJourneys(
+      phoneNumber,
+      UserCommunicationJourneyType.BULK_SMS,
+    )
+
+    const message = await sendSMS({
+      body,
+      to: phoneNumber,
+    })
+
+    if (message) {
+      await createCommunication(communicationJourneys, message.sid)
+    }
+
+    return message
+  })
+
+  const failedPhoneNumbers: string[] = []
+
+  let messagesSent = 0
+  await Promise.allSettled(enqueueMessagesPromise).then(results => {
+    results.forEach(result => {
+      if (result.status === 'rejected') {
+        if (result.reason) {
+          if (result.reason.code === 20429 || result.reason.code === 'ECONNABORTED') {
+            // Too Many Requests
+            if ('phoneNumber' in result.reason) {
+              failedPhoneNumbers.push(result.reason.phoneNumber)
+            }
+          } else if (result.reason.code === 21211) {
+            // Invalid phone number
+            // TODO: flag users with this phone number
+          } else {
+            console.log(result.reason)
+            Sentry.captureException('Unexpected Twilio error', {
+              extra: { reason: result.reason },
+              tags: {
+                domain: 'bulkSMS',
+              },
+            })
+          }
+        } else {
+          Sentry.captureException('Twilio failed with no reason', {
+            extra: {
+              reason: result.reason,
+            },
+            tags: {
+              domain: 'bulkSMS',
+            },
+          })
+        }
+      } else {
+        messagesSent += 1
+      }
+    })
+  })
+
+  // exponential backoff retry
+  if (failedPhoneNumbers.length > 0) {
+    await sleep(10000 * attempt)
+
+    messagesSent += await enqueueMessages(failedPhoneNumbers, body, attempt + 1)
+  }
+
+  return messagesSent
+}
+
+function formatTime(seconds: number) {
+  if (seconds < 60) {
+    return `${seconds.toPrecision(2)} seconds`
+  } else if (seconds < 3600) {
+    // less than 60 minutes (3600 seconds)
+    const minutes = Math.ceil(seconds / 60)
+    return `${minutes} minutes`
+  } else if (seconds < 86400) {
+    // less than 24 hours (86400 seconds)
+    const hours = Math.ceil(seconds / 3600)
+    return `${hours} hours`
+  } else {
+    // 24 hours or more
+    const days = Math.ceil(seconds / 86400)
+    return `${days} days`
+  }
```

Comment:
We have a seconds duration shared constant for this kind of comparison
```suggestion
  if (seconds < SECONDS_DURATION.MINUTE) {
    return `${seconds.toPrecision(2)} seconds`
  } else if (seconds < SECONDS_DURATION.HOUR) {
    // less than 60 minutes (3600 seconds)
    const minutes = Math.ceil(seconds / SECONDS_DURATION.MINUTE)
    return `${minutes} minutes`
  } else if (seconds < SECONDS_DURATION.DAY) {
    // less than 24 hours (86400 seconds)
    const hours = Math.ceil(seconds / SECONDS_DURATION.HOUR)
    return `${hours} hours`
  } else {
    // 24 hours or more
    const days = Math.ceil(seconds / SECONDS_DURATION.DAY)
    return `${days} days`
  }
```